In [1]:
import pandas as pd
from scipy import stats
import numpy as np

### Задача 1  
Даны две независимые выборки. Не соблюдается условие нормальности  
x1 380,420, 290  
y1 140,360,200,900  
Сделайте вывод по результатам, полученным с помощью функции   





#### Применяем критерий Манна-Уитни

In [2]:
x1 = np.array([380,420, 290])  
y1 = np.array([140,360,200,900])
stats.mannwhitneyu(x1, y1)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

#### Вывод: смотря на результат p-value (0.6) а это больше чем альфа (5%), можем сделать вывод в пользу нулевой гипотезы. Различий нет.

### Задача 2
Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть ли статистически значимые различия?

1е измерение до приема препарата: 150, 160, 165, 145, 155  
2е измерение через 10 минут: 140, 155, 150, 130, 135  
3е измерение через 30 минут: 130, 130, 120, 130, 125


#### Применяем критерий Фридмана

In [3]:
before_pill = np.array([150, 160, 165, 145, 155])
after_pill_1 =np.array([140, 155, 150, 130, 135])
after_pill_2 =np.array([130, 130, 120, 130, 125])
stats.friedmanchisquare(before_pill, after_pill_1, after_pill_2 )

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

#### Вывод: смотря на результат p-value (0.008) а это меньше знчания  альфа (5%), можем сделать вывод в пользу альтернативной гипотезы. Различия есть.

### Задача 3
Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.


#### Здесь применяем критерий Уилкоксона так как идет сравнение двух зависимых групп

In [8]:
before = np.array([150, 160, 165, 145, 155])
after = np.array([140, 155, 150, 130, 135])
before - after

array([10,  5, 15, 15, 20])

In [6]:
stats.wilcoxon(before, after )

WilcoxonResult(statistic=0.0, pvalue=0.0625)

#### Вывод: смотря на результат p-value (0.0625) а это больше чем альфа (5%), можем сделать вывод в пользу нулевой гипотезы. Различий нет.

### Задача 4
Даны 3 группы учеников плавания.   
В 1 группе время на дистанцию 50 м составляют:  
56, 60, 62, 55, 71, 67, 59, 58, 64, 67  
Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53  
Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54



#### применяем критерий Крускала-Уоллиса

In [9]:
group_1 = np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
group_2 = np.array([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
group_3 = np.array([57, 67, 49, 48, 47, 55, 66, 51, 54])

stats.kruskal(group_1, group_2, group_3)


KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)

#### Вывод: смотря на результат p-value (0.065) а это больше чем альфа (5%), можем сделать вывод в пользу нулевой гипотезы. Различий нет.